### PR TITLE
Fully support RAFS blobs with inlined bootstrap

### PR DESCRIPTION
**Pull request by jiangliu**
RAFS metadata/bootstrap can be inlined into RAFS data blobs with `nydus-image create --blob-inline-meta`.
So we need update `nydus-image` and `nydusd` to fully support inlined metadata.
This PR enhances following components to support inlined metadata:
- nydus-image create --blob-inline-meta --features blob-toc
- nydus-image create --parent-bootstrap
- nydus-image create --chunk-dict
- nydus-image check
- nydus-image inspect
- nydus-image stat
- nydus-image merge
- nydusd

There are still several TODOs:
- support images generated by `nydus-image create --blob-inline-meta` without `--features blob-toc`. Currently it only supports images generated by `nydus-image create --blob-inline-meta --features blob-toc`
- --chunk-dict doesn't work with ZRan images
- --blob-id conflicts with ZRan and `--blob-inlined-meta`

### CODE REVIEW

**Comment by jiangliu** [comment]
"--chunk-dict doesn't work with ZRan images", this issue has been fixed.

File: src/bin/nydus-image/builder/mod.rs [hunk 1]
```diff
@@ -71,10 +71,24 @@ fn dump_bootstrap(
...
+        // Make sure blob id is updated according to blob hash if not specified by user.
         if blob_ctx.blob_id.is_empty() {
+            // `Blob::dump()` should have set `blob_ctx.blob_id` to referenced OCI tarball for
+            // ref-type conversion.
             assert!(!ctx.conversion_type.is_to_ref());
-            // Make sure blob id is updated according to blob hash if user not specified.
-            blob_ctx.blob_id = format!("{:x}", blob_ctx.blob_hash.clone().finalize());
+            if ctx.blob_inline_meta {
+                // Set special blob id for blob with inlined meta.
+                blob_ctx.blob_id = "x".repeat(64);
```
Comment (imeoer) [inline src/bin/nydus-image/builder/mod.rs:81]: It seems the blob id `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx` will be persisted to bootstrap in the merge command.

**Comment by imeoer** [comment]
https://github.com/containerd/nydus-snapshotter/blob/main/tests/converter_test.go#L416

Maybe we should ensure the snapshotter converter tests still can pass like:

```
env NYDUS_NYDUSD=/path/to/nydusd-master \
NYDUS_BUILDER=/path/to/nydus-image-master \
NYDUS_NYDUSD_OLD=/path/to/nydusd-v2.1 \
NYDUS_BUILDER_OLD=/path/to/nydus-image-v2.1 \
go test -v -count=1 -run TestPack ./tests
```

**Comment by imeoer** [comment]
@loheagn The PR has added `--config` option for `nydus-image`, maybe we can implement a new `content store proxy` backend in https://github.com/dragonflyoss/image-service/pull/974.

**Comment by imeoer** [comment]
@jiangliu Wait to merge until we can pass the converter test: https://github.com/containerd/nydus-snapshotter/blob/main/tests/converter_test.go#L416.

**Comment by jiangliu** [comment]
"support images generated by nydus-image create --blob-inline-meta without --features blob-toc", got fixed too.

**Review by imeoer — APPROVED** [review]

**Comment by imeoer** [comment]
Let's move forward first.

**Pull request merged**